### PR TITLE
CMake: Introduce a dealii:: namespace for targets and harmonize naming

### PR DESCRIPTION
**Pull request by tamiko**
The top-level targets are now named dealii_debug and dealii_release (and
their exported counterparts dealii::dealii_debug and
dealii::dealii_release).

### CODE REVIEW

**Comment by tamiko** [comment]
I need this as a preparatory step to convert our feature configuration to interface targets.

File: cmake/setup_deal_ii.cmake [hunk 1]
```diff
@@ -100,7 +102,11 @@ set(DEAL_II_VERSION ${DEAL_II_VERSION_MAJOR}.${DEAL_II_VERSION_MINOR}.${DEAL_II_
 #                                                                      #
 ########################################################################
 
-set(DEAL_II_PROJECT_CONFIG_NAME "${DEAL_II_PACKAGE_NAME}")
+string(REPLACE "." "" _namespace "${DEAL_II_PACKAGE_NAME}")
+string(TOLOWER "${_namespace}" _namespace)
+set_if_empty(DEAL_II_NAMESPACE "${_namespace}")
```
Comment (masterleinad) [inline cmake/setup_deal_ii.cmake:107]: Is there a good reason to not just use `dealii` for `DEAL_II_NAMESPACE`?

Comment (tamiko) [reply cmake/setup_deal_ii.cmake:107]: The rest of the file derives all names from `${DEAL_II_PACKAGE_NAME}`, thus the small dance to derive "dealii" from it.

**Review by masterleinad — APPROVED** [review]

**Pull request merged**